packages/account/src/Sections/Security/ApiToken/api-token.tsx

```diff
@@ -1,4 +1,306 @@
+            ...prev_state,
+            ...value,
+        }),
+        {
+            api_tokens: [],
+            is_loading: true,
+            is_success: false,
+            error_message: '',
+            show_delete: false,
+            is_delete_loading: false,
+            is_delete_success: false,
+        }
+    );
+
+    const timeout_ref = React.useRef<NodeJS.Timeout | undefined>();
+
+    React.useEffect(() => {
+        getApiTokens();
+
+        return () => {
+            clearTimeout(timeout_ref.current);
+        };
+    }, []);
+
+    React.useEffect(() => {
+        if (prev_is_switching.current !== is_switching) {
+            prev_is_switching.current = is_switching;
+            getApiTokens();
+        }
+    }, [is_switching]);
+
+    const initial_form: TApiTokenForm = {
+        token_name: '',
+        read: false,
+        trade: false,
+        payments: false,
+        trading_information: false,
+        admin: false,
+    };
+
+    const validateFields = (values: TApiTokenForm) => {
+        const errors: FormikErrors<TApiTokenForm> = {};
+        const token_name = values?.token_name?.trim();
+
+        if (!token_name) {
+            errors.token_name = localize('Please enter a token name.');
+        } else if (!/^[A-Za-z0-9\s_]+$/g.test(token_name)) {
+            errors.token_name = localize('Only letters, numbers, and underscores are allowed.');
+        } else if (token_name.length < TOKEN_LIMITS.MIN) {
+            errors.token_name = localize(
+                'Length of token name must be between {{MIN_TOKEN}} and {{MAX_TOKEN}} characters.',
+                {
+                    MIN_TOKEN: TOKEN_LIMITS.MIN,
+                    MAX_TOKEN: TOKEN_LIMITS.MAX,
+                }
+            );
+        } else if (token_name.length > TOKEN_LIMITS.MAX) {
+            errors.token_name = localize('Maximum {{MAX_TOKEN}} characters.', { MAX_TOKEN: TOKEN_LIMITS.MAX });
+        }
+
+        return errors;
+    };
+
+    const selectedTokenScope = (values: TApiTokenForm) =>
+        Object.keys(values).filter(
+            item => item !== 'token_name' && Boolean(values[item as keyof TApiTokenForm])
+        ) as NonNullable<NonNullable<TApitoken['tokens']>[0]['scopes']>;
+
+    const handleSubmit = async (
+        values: TApiTokenForm,
+        { setSubmitting, setFieldError, resetForm }: FormikHelpers<TApiTokenForm>
+    ) => {
+        const token_response = await WS.apiToken({
+            api_token: 1,
+            new_token: values.token_name,
+            new_token_scopes: selectedTokenScope(values),
+        });
+        if (token_response.error) {
+            setFieldError('token_name', token_response.error.message);
+        } else {
+            setState({
+                is_success: true,
+                api_tokens: getPropertyValue(token_response, ['api_token', 'tokens']),
+            });
+            setTimeout(() => {
+                setState({ is_success: false });
+            }, 500);
```

Comment:
Please add a cleanup function for timeout, else it may create a memory leak error some place